source/solvers/postprocessing_cfd.cc

```diff
@@ -54,72 +54,112 @@ calculate_pressure_drop(const DoFHandler<dim> &       dof_handler,
                                    update_values | update_quadrature_points |
                                      update_normal_vectors | update_JxW_values);
 
+  const FEValuesExtractors::Vector velocities(0);
   const FEValuesExtractors::Scalar pressure(dim);
 
-  const unsigned int n_q_points      = cell_quadrature_formula.size();
   const unsigned int face_n_q_points = face_quadrature_formula.size();
 
-  std::vector<double> present_pressure_values(n_q_points);
+  std::vector<Tensor<1, dim>> velocity_values(face_n_q_points);
+  std::vector<double>         pressure_values(face_n_q_points);
+
+  double static_pressure_outlet_boundary = 0.;
+  double static_pressure_inlet_boundary  = 0.;
+  double static_pressure_drop            = 0.;
 
-  double pressure_outlet_boundary = 0;
-  double outlet_surface           = 0;
-  double pressure_inlet_boundary  = 0;
-  double inlet_surface            = 0;
-  double pressure_drop            = 0;
+  double dynamic_pressure_outlet_boundary = 0.;
+  double dynamic_pressure_inlet_boundary  = 0.;
+  double dynamic_pressure_drop            = 0.;
+
+  double outlet_surface = 0.;
+  double inlet_surface  = 0.;
+
+  double temp_surface                      = 0.;
+  double temp_static_pressure_at_boundary  = 0.;
+  double temp_dynamic_pressure_at_boundary = 0.;
 
   for (const auto &cell : dof_handler.active_cell_iterators())
     {
       if (cell->is_locally_owned())
         {
-          fe_values.reinit(cell);
-          // Gather pressure (values)
-          fe_values[pressure].get_function_values(evaluation_point,
-                                                  present_pressure_values);
-
-          for (unsigned int q = 0; q < face_n_q_points; ++q)
+          for (unsigned int face = 0; face < GeometryInfo<dim>::faces_per_cell;
+               face++)
             {
-              for (const auto &face : cell->face_iterators())
+              if (cell->face(face)->at_boundary())
```

Comment:
here I would also check if it's the inlet or the outlet using the face_id. Otherwise you are doing calculations for no reasons no?